Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -188,14 +188,60 @@ internal Status GetStatus()
             {
                 var grpcStatus = OriginalResponseMessage.StatusCode switch
                 {
-                    HttpStatusCode.OK => StatusCode.OK,
-                    HttpStatusCode.Unauthorized => StatusCode.Unauthenticated,
-                    HttpStatusCode.Forbidden => StatusCode.PermissionDenied,
+                    // Note: this switch statement contains all the unique named values within HttpStatusCode (as of 2021-08-09).
+
+                    HttpStatusCode.Accepted => StatusCode.OK,
+                    HttpStatusCode.Ambiguous => StatusCode.Unknown,
+                    HttpStatusCode.BadGateway => StatusCode.InvalidArgument,
                     HttpStatusCode.BadRequest => StatusCode.InvalidArgument,
+                    // Note: this is ambiguous between FailedPrecondition, Aborted or Unavailable.
+                    HttpStatusCode.Conflict => StatusCode.FailedPrecondition,
+                    HttpStatusCode.Continue => StatusCode.Unknown, // We don't expect to get this
+                    HttpStatusCode.Created => StatusCode.OK, // In the 200 range
+                    HttpStatusCode.ExpectationFailed => StatusCode.FailedPrecondition,
+                    HttpStatusCode.Forbidden => StatusCode.PermissionDenied,
+                    // Note: Found and Redirect are both 302
+                    HttpStatusCode.GatewayTimeout => StatusCode.DeadlineExceeded,
+                    HttpStatusCode.Gone => StatusCode.NotFound,
+                    HttpStatusCode.HttpVersionNotSupported => StatusCode.Unknown,
```

Comment:
Should this one be InvalidArgument?

Reply:
I generally expect InvalidArgument to mean that there's something the caller can do about it - and that wouldn't be the case here.

Reply:
(Mind you, it's not the case in various other things either...)